src/components/BalanceService.tsx

```diff
@@ -0,0 +1,23 @@
+import MemoryService from "../core/Memory";
+
+class BalanceService {
+    saveValues: (balances: { date: Date, value: number }[]) => void;
+    balances: { date: Date, value: number }[] ;
+
+    addBalance(valueFromBalanceForm: number) {
+        let valuesDatesFromMemory = MemoryService.instance.getValues();
+        let valueWithDate = {date: new Date(), value: valueFromBalanceForm};
+        let valueToSave = [valueWithDate, ...valuesDatesFromMemory];
```

Comment:
I don't think you need to create another variable here.

Reply:
Do you mean: "valueToSave"? And use balances from BalanceService instead?

Reply:
yup.
saveValues shouldnt mutate passed array.

Reply:
ok